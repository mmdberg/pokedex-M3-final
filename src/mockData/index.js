export const mockPokemonTypeData = [
  {
    "id": "1",
    "name": "normal",
    "pokemon": [
      "16",
      "17",
      "18",
      "19",
      "20"
    ]
  },
]