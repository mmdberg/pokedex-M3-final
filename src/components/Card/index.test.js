import { Card } from './index';
import { shallow } from 'enzyme';
import React from 'react';
import { mockType } from '../../mockData';

describe('Card', () => {
  it('should match snapshot without active pokemon', () => {
    const wrapper = shallow(<Card type={mockType} pokemon={[]}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should match snapshot with active pokemon', () => {
    const mockPokemon = [{
      name: 'spearow',
      weight: 20,
      sprites: {back_default: 'img'},
      type: "3"
    }]
    const wrapper = shallow(<Card type={mockType} pokemon={mockPokemon}/>)
    expect(wrapper).toMatchSnapshot()
  });
});