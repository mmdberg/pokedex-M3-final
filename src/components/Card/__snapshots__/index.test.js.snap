// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with active pokemon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    pokemon={
      Array [
        Object {
          "name": "spearow",
          "sprites": Object {
            "back_default": "img",
          },
          "type": "3",
          "weight": 20,
        },
      ]
    }
    type={
      Object {
        "id": "1",
        "name": "normal",
        "pokemon": Array [
          "16",
          "17",
          "18",
          "19",
          "20",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          normal
        </h2>,
        <div>
          <div>
            <h4>
              spearow
            </h4>
            <p>
              weight: 
              20
               lbs
            </p>
            <p>
              type: 
              3
            </p>
            <img
              alt=""
              src="img"
            />
          </div>
        </div>,
      ],
      "className": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "normal",
        },
        "ref": null,
        "rendered": "normal",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h4>
                spearow
              </h4>
              <p>
                weight: 
                20
                 lbs
              </p>
              <p>
                type: 
                3
              </p>
              <img
                alt=""
                src="img"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  spearow
                </h4>,
                <p>
                  weight: 
                  20
                   lbs
                </p>,
                <p>
                  type: 
                  3
                </p>,
                <img
                  alt=""
                  src="img"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "spearow",
                },
                "ref": null,
                "rendered": "spearow",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "weight: ",
                    20,
                    " lbs",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "weight: ",
                  20,
                  " lbs",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "type: ",
                    "3",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "type: ",
                  "3",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "img",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            normal
          </h2>,
          <div>
            <div>
              <h4>
                spearow
              </h4>
              <p>
                weight: 
                20
                 lbs
              </p>
              <p>
                type: 
                3
              </p>
              <img
                alt=""
                src="img"
              />
            </div>
          </div>,
        ],
        "className": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "normal",
          },
          "ref": null,
          "rendered": "normal",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h4>
                  spearow
                </h4>
                <p>
                  weight: 
                  20
                   lbs
                </p>
                <p>
                  type: 
                  3
                </p>
                <img
                  alt=""
                  src="img"
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    spearow
                  </h4>,
                  <p>
                    weight: 
                    20
                     lbs
                  </p>,
                  <p>
                    type: 
                    3
                  </p>,
                  <img
                    alt=""
                    src="img"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "spearow",
                  },
                  "ref": null,
                  "rendered": "spearow",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "weight: ",
                      20,
                      " lbs",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "weight: ",
                    20,
                    " lbs",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "type: ",
                      "3",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "type: ",
                    "3",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "img",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot without active pokemon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    pokemon={Array []}
    type={
      Object {
        "id": "1",
        "name": "normal",
        "pokemon": Array [
          "16",
          "17",
          "18",
          "19",
          "20",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          normal
        </h2>,
        false,
      ],
      "className": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "normal",
        },
        "ref": null,
        "rendered": "normal",
        "type": "h2",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            normal
          </h2>,
          false,
        ],
        "className": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "normal",
          },
          "ref": null,
          "rendered": "normal",
          "type": "h2",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
